cases/views.py

```diff
@@ -561,7 +561,7 @@ def done(self, form_list, form_dict, **kwargs):
         send_emails(self.request, complaint, "reoccurrence")
         # Reopen a case if it was closed - do this after email so email can know if it was closed
         if self.object.closed:
-            self.object.closed = False
+            self.object.reopen()
```

Comment:
This will mean the last action on the case is always the case reopening, rather than the complaint, which isn't ideal, sort of a duplicate. I understand the reason why you've done this, as discussed, but I worry this is overcomplicating things and might lead to further confusion elsewhere we've (well, more likely I've!) not thought of. This is my fault for suggesting editing the type of an action without thinking through all the consequences, as you've then been doing. If they close a case by mistake with an action, they can reopen it with another action - that feels like a better way to go than to let them edit the action to no longer be a closed action. If they do another wrong type, they could update the text to explain that. So I'm leaning towards not allowing type to be edited (or at least not allowing closed/reopened), so we don't then need to have these complications with the closed/reopen. Sorry, happy to discuss tomorrow.

Reply:
Nice, this makes a lot of sense to me, will make it so. I definitely had tunnel-vision on getting type editing to work, iI probably should have checked what was I doing against the scope when implementation was getting quite complicated, lesson learnt.